libobs/obs-cocoa.m

```diff
@@ -177,6 +183,247 @@ static bool dstr_from_cfstring(struct dstr *str, CFStringRef ref)
+    [OBS_KEY_NUMPERIOD]   = {.desc = ". (Keypad)", .is_valid = true},
+    [OBS_KEY_NUMSLASH]    = {.desc = "/ (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM0]        = {.desc = "0 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM1]        = {.desc = "1 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM2]        = {.desc = "2 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM3]        = {.desc = "3 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM4]        = {.desc = "4 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM5]        = {.desc = "5 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM6]        = {.desc = "6 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM7]        = {.desc = "7 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM8]        = {.desc = "8 (Keypad)", .is_valid = true},
+    [OBS_KEY_NUM9]        = {.desc = "9 (Keypad)", .is_valid = true},
+    [OBS_KEY_MOUSE1]      = {.desc = "Mouse 1",    .is_valid = true},
+    [OBS_KEY_MOUSE2]      = {.desc = "Mouse 2",    .is_valid = true},
+    [OBS_KEY_MOUSE3]      = {.desc = "Mouse 3",    .is_valid = true},
+    [OBS_KEY_MOUSE4]      = {.desc = "Mouse 4",    .is_valid = true},
+    [OBS_KEY_MOUSE5]      = {.desc = "Mouse 5",    .is_valid = true},
+    [OBS_KEY_MOUSE6]      = {.desc = "Mouse 6",    .is_valid = true},
+    [OBS_KEY_MOUSE7]      = {.desc = "Mouse 7",    .is_valid = true},
+    [OBS_KEY_MOUSE8]      = {.desc = "Mouse 8",    .is_valid = true},
+    [OBS_KEY_MOUSE9]      = {.desc = "Mouse 9",    .is_valid = true},
+    [OBS_KEY_MOUSE10]     = {.desc = "Mouse 10",   .is_valid = true},
+    [OBS_KEY_MOUSE11]     = {.desc = "Mouse 11",   .is_valid = true},
+    [OBS_KEY_MOUSE12]     = {.desc = "Mouse 12",   .is_valid = true},
+    [OBS_KEY_MOUSE13]     = {.desc = "Mouse 13",   .is_valid = true},
+    [OBS_KEY_MOUSE14]     = {.desc = "Mouse 14",   .is_valid = true},
+    [OBS_KEY_MOUSE15]     = {.desc = "Mouse 15",   .is_valid = true},
+    [OBS_KEY_MOUSE16]     = {.desc = "Mouse 16",   .is_valid = true},
+    [OBS_KEY_MOUSE17]     = {.desc = "Mouse 17",   .is_valid = true},
+    [OBS_KEY_MOUSE18]     = {.desc = "Mouse 18",   .is_valid = true},
+    [OBS_KEY_MOUSE19]     = {.desc = "Mouse 19",   .is_valid = true},
+    [OBS_KEY_MOUSE20]     = {.desc = "Mouse 20",   .is_valid = true},
+    [OBS_KEY_MOUSE21]     = {.desc = "Mouse 21",   .is_valid = true},
+    [OBS_KEY_MOUSE22]     = {.desc = "Mouse 22",   .is_valid = true},
+    [OBS_KEY_MOUSE23]     = {.desc = "Mouse 23",   .is_valid = true},
+    [OBS_KEY_MOUSE24]     = {.desc = "Mouse 24",   .is_valid = true},
+    [OBS_KEY_MOUSE25]     = {.desc = "Mouse 25",   .is_valid = true},
+    [OBS_KEY_MOUSE26]     = {.desc = "Mouse 26",   .is_valid = true},
+    [OBS_KEY_MOUSE27]     = {.desc = "Mouse 27",   .is_valid = true},
+    [OBS_KEY_MOUSE28]     = {.desc = "Mouse 28",   .is_valid = true},
+    [OBS_KEY_MOUSE29]     = {.desc = "Mouse 29",   .is_valid = true},
+};
+
+static const macOS_glyph_desc_t key_glyphs[0xFF] = {
```

Comment:
Is there a reason this array has size 255?

Reply:
Probably an oversight, using `OBS_KEY_LAST_VALUE` should work here as well.

Reply:
Ah wait, I'm mistaken - that specific array uses the macOS key codes as its key, so there is no single "largest value" (like `OBS_KEY_LAST_VALUE`) that could be used here.

In theory one could use `0x7F` here (as `0x7E` seems to be the highest key code defined in Carbon's `Events.h` (and that header hasn't changed in ages), `0xFF` is ensured to encapsulate all current key codes though.

Reply:
Yeah I suppose that's fine. I guess a 0x7E minimally sized array would take up a bit less space, but unlikely to matter. Regardless I think either a brief code comment or a named constant for the size might be good here, just to make it easier to understand at a glance.

Reply:
I use the keyCode bit mask now (which is 8 bits wide) and downshift it by 8 bytes to effectively get `0xFF`, but it should make it more obvious (as that implies that a keyCode _has_ to fit into the width of the mask for a Carbon event).